mitmproxy/proxy/protocol/websocket.py

```diff
@@ -70,43 +71,50 @@ def _handle_data_frame(self, frame, source_conn, other_conn, is_server):
             self.flow.messages.append(websocket_message)
             self.channel.ask("websocket_message", self.flow)
 
-            def get_chunk(payload):
-                if len(payload) == length:
-                    # message has the same length, we can reuse the same sizes
-                    pos = 0
-                    for s in original_chunk_sizes:
-                        yield payload[pos:pos + s]
-                        pos += s
+            if not self.flow.stream:
+                def get_chunk(payload):
+                    if len(payload) == length:
+                        # message has the same length, we can reuse the same sizes
+                        pos = 0
+                        for s in original_chunk_sizes:
+                            yield payload[pos:pos + s]
+                            pos += s
+                    else:
+                        # just re-chunk everything into 10kB frames
+                        chunk_size = 10240
```

Comment:
Minor suggestion: Make this a power of two, see e.g. the note in https://docs.python.org/3/library/socket.html#socket.socket.recv.